spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/config/GoogleConfigEnvironment.java

```diff
@@ -38,6 +40,9 @@
 	}
 
 	List<Variable> getVariables() {
+		if (variables == null) {
```

Comment:
Assuming that `getVariables()` is called more frequently than `setVariables()`, I think I would rather do it in the setter. I guess it's a stylistic thing though.